ggml-cuda.cu

```diff
@@ -1225,7 +1225,7 @@ static void ggml_cuda_op_mul_mat_cublas(
 
     // the main device has a larger memory buffer to hold the results from all GPUs
     // ldc == nrows of the matrix that cuBLAS writes into
-    int64_t ldc = id == ctx.device ? ne0 : row_diff;
+    int ldc = id == ctx.device ? ne0 : row_diff;
```

Comment:
Wait, why is this being changed? I thought the problem was that certain ints had too few bits for large models.

Reply:
Did you maybe, in response to one of my earlier comments, accidentally change more places than just the ones originally touched in this PR?

Reply:
@JohannesGaessler This one was reverted following an earlier comment questionning why it was changed in the first place. As previously mentioned, I have limited knowledge about these vars and rely on others expertise for the review. And because of the large number of ints that was overflowing, I had to guess and change them in batches until all the crashes were fixed, but surely I most likely changed more than needed.

Reply:
It's fine to change more `int` to `int64_t` than necessary. But this is a change where a value was `int64_t` on master to `int` with your PR. I think this was done on accident when you reverted some of your other changes.

Reply:
That is because my previous PR was merged into master, this is a subsequent PR. I can revert them back if needed.

Reply:
The revert is in a single commit https://github.com/dranger003/llama.cpp/commit/9acb43d7fa0b8da867570c975d33f0728951ca46 so if these are all fine I can delete that one commit.

Reply:
Just delete the commit I'd say. Using `int64_t` has no disadvantages other than maybe slightly worse performance and I was not able to measure any performance difference whatsoever.

Reply:
I pushed a rebase to remove the revert commit.

Reply:
It seems this particular change is still there. Revert it and I'll merge.

Reply:
`cublasGemmEx` takes an `int` anyway, so this doesn't really matter. There is a 64-bit interface to cublas, but I don't think there are any cases where a single dimension is larger than 2^31-1.